shell/platform/android/test/io/flutter/plugin/platform/PlatformPluginTest.java

```diff
@@ -61,6 +65,19 @@ public void platformPlugin_getClipboardData() {
     ClipData clip = ClipData.newPlainText("label", "Text");
     clipboardManager.setPrimaryClip(clip);
     assertNotNull(platformPlugin.mPlatformMessageHandler.getClipboardData(clipboardFormat));
+
+    ContentResolver contentResolver = RuntimeEnvironment.application.getContentResolver();
+    Uri uri = Uri.parse("content://media/external_primary/images/media/");
+    clip = ClipData.newUri(contentResolver, "URI", uri);
+    clipboardManager.setPrimaryClip(clip);
+    assertNull(platformPlugin.mPlatformMessageHandler.getClipboardData(clipboardFormat));
+
+    uri =
+        RingtoneManager.getActualDefaultRingtoneUri(
+            RuntimeEnvironment.application.getApplicationContext(), RingtoneManager.TYPE_RINGTONE);
+    clip = ClipData.newUri(contentResolver, "URI", uri);
+    clipboardManager.setPrimaryClip(clip);
+    assertNotNull(platformPlugin.mPlatformMessageHandler.getClipboardData(clipboardFormat));
```

Comment:
Can we do better than assertNotNull here? Maybe assert the data is the string we expect?

Reply:
@danafergan 

It's not possible, The Uri will be converted to InputStreamReader in `coerceToText` function: 

```
 if (uri != null) {
                // First see if the URI can be opened as a plain text stream
                // (of any sub-type).  If so, this is the best textual
                // representation for it.
                final ContentResolver resolver = context.getContentResolver();
                AssetFileDescriptor descr = null;
                FileInputStream stream = null;
                InputStreamReader reader = null;
                try {
                    try {
                        // Ask for a stream of the desired type.
                        descr = resolver.openTypedAssetFileDescriptor(uri, "text/*", null);
                    } catch (SecurityException e) {
                        Log.w("ClipData", "Failure opening stream", e);
                    } catch (FileNotFoundException|RuntimeException e) {
                        // Unable to open content URI as text...  not really an
                        // error, just something to ignore.
                    }
                    if (descr != null) {
                        try {
                            stream = descr.createInputStream();
                            reader = new InputStreamReader(stream, "UTF-8");

                            // Got it...  copy the stream into a local string and return it.
                            final StringBuilder builder = new StringBuilder(128);
                            char[] buffer = new char[8192];
                            int len;
                            while ((len=reader.read(buffer)) > 0) {
                                builder.append(buffer, 0, len);
                            }
                            return builder.toString();
                        } catch (IOException e) {
                            // Something bad has happened.
                            Log.w("ClipData", "Failure loading text", e);
                            return e.toString();
                        }
                    }
                } finally {
                    IoUtils.closeQuietly(descr);
                    IoUtils.closeQuietly(stream);
                    IoUtils.closeQuietly(reader);
                }

                // If we couldn't open the URI as a stream, use the URI itself as a textual
                // representation (but not for "content", "android.resource" or "file" schemes).
                final String scheme = uri.getScheme();
                if (SCHEME_CONTENT.equals(scheme)
                        || SCHEME_ANDROID_RESOURCE.equals(scheme)
                        || SCHEME_FILE.equals(scheme)) {
                    return "";
                }
                return uri.toString();
            }
```

Reply:
Can't we just read the inputstream and make sure it contains what we'd expect?

Reply:
@dnfield  wdyt : 

<details>
  <summary>assertEquals</summary>

```
..............
..............

uri = RingtoneManager.getActualDefaultRingtoneUri...........

String uriData = platformPlugin.mPlatformMessageHandler.getClipboardData(clipboardFormat).toString();

assertEquals(getUriData(uri, resolver), uriData);

String getUriData(Uri uri, ContentResolver resolver) throws IOException {
    AssetFileDescriptor descr = null;
    FileInputStream stream = null;
    InputStreamReader reader = null;
    try {
      try {
        descr = resolver.openTypedAssetFileDescriptor(uri, "text/*", null);
      } catch (SecurityException e) {
        return null;
      } catch (FileNotFoundException | RuntimeException e) {
        return null;
      }
      if (descr != null) {
        try {
          stream = descr.createInputStream();
          reader = new InputStreamReader(stream, "UTF-8");

          final StringBuilder builder = new StringBuilder(128);
          char[] buffer = new char[8192];
          int len;
          while ((len = reader.read(buffer)) > 0) {
            builder.append(buffer, 0, len);
          }
          return builder.toString();
        } catch (IOException e) {
          return e.toString();
        }
      }
    } finally {
      if (descr != null) descr.close();
      if (stream != null) stream.close();
      if (reader != null) reader.close();
    }

    // If we couldn't open the URI as a stream, use the URI itself as a textual
    // representation (but not for "content", "android.resource" or "file" schemes).
    final String scheme = uri.getScheme();
    if (SCHEME_CONTENT.equals(scheme)
        || SCHEME_ANDROID_RESOURCE.equals(scheme)
        || SCHEME_FILE.equals(scheme)) {
      return "";
    }
    return uri.toString();
  }
```
</details>

Reply:
Sure.  If there's something slightly less complicated where we can just check that it's an InputStream and that its contents are sane that's probably fine too.

Reply:
I will test with another Build Tools version and I will come back with feedback

Reply:
@dnfield Test Updated. Thanks